src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectProperty.java

```diff
@@ -181,30 +234,147 @@ public String getPropertyName() {
-                throws hudson.model.Descriptor.FormException {
-            if(formData == null || formData.isNullObject()) {
-                return null;
-            }
-            JSONObject form = formData.getJSONObject(getPropertyName());
-            if(form == null || form.isNullObject()) {
-                return null;
+        public String getIconFileName() {
+            return ProjectQueueItemAuthenticator.isConfigured() ? "secure.png" : null;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getDisplayName() {
+            return "Authorization";
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getUrlName() {
+            return "authorization";
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getIconClassName() {
+            return ProjectQueueItemAuthenticator.isConfigured() ? "icon-secure" : null;
+        }
+
+        /**
+         * Handles the submisson of the authorization configuration.
+         * @param req the request.
+         * @return the response.
+         * @throws IOException when things go wrong.
+         * @throws ServletException when things go wrong.
+         */
+        @RequirePOST
+        @NonNull
+        @Restricted(NoExternalUse.class)
+        @SuppressWarnings("unused") // stapler web method binding
+        public synchronized HttpResponse doAuthorize(@NonNull StaplerRequest req) throws IOException, ServletException {
+            job.checkPermission(Job.CONFIGURE);
+            JSONObject json = req.getSubmittedForm();
+            JSONObject o = json.optJSONObject(getPropertyDescriptor().getJsonSafeClassName());
+            AuthorizeProjectProperty submitted = o != null ? req.bindJSON(AuthorizeProjectProperty.class, o) : null;
+            // now it is safe to make the changes
+            BulkChange bc = new BulkChange(job);
+            try {
+                AuthorizeProjectProperty existing = getProperty();
+                if (existing != null) {
+                    job.removeProperty(existing);
+                }
+                if (submitted != null) {
+                    job.addProperty(submitted);
+                }
+                job.save();
+                bc.commit();
+                return FormApply.success("../");
+            } catch (IOException e){
+                bc.abort();
+                throw e;
             }
-            
-            AuthorizeProjectStrategy strategy = AuthorizeProjectUtil.bindJSONWithDescriptor(req, form, "strategy", AuthorizeProjectStrategy.class);
-            
-            return new AuthorizeProjectProperty(strategy);
+        }
+
+    }
+
+    /**
+     * The action factory responsible for adding the {@link AuthorizationAction}.
+     *
+     * @since 1.3.0
+     */
+    @Extension(ordinal = Double.MAX_VALUE / 2)

```

Comment:
We want the action close to the top... but we do t want to be at the top in case there is some need to put something above